### ISSUE TITLE
Some players report the game "freezing" upon death

### ISSUE THREAD

**Issue by cmbasnett** [DarklightGames/DarkestHour#880]
Not sure what the deal is here. Doesn't seem to be dependent on OS. A very small contingent of players experienced this. It could quite possibly be related to the spectator mode changes I did, since the player reporting it say it started happening when we did the the October 15th test, which was the first time the spectator mode changes were implemented. I will investigate!

**Comment by cmbasnett** [comment]
We have some video evidence of the problem:

https://www.youtube.com/watch?v=WR7GEGh6Ndk

At the 24:00 mark, he soon gets into a "gray screen" and the frame-rate starts to tank until he eventually spawns and it fixes it.

**Comment by cmbasnett** [comment]
This bug is likely a result of one of the changes made regarding #803.

**Comment by AndrewTheel** [comment]
We need as much feedback as possible and as often as possible.

**Comment by cmbasnett** [comment]
I will have to go over the spectator changes made during the 8.0 alpha and
see if there are any places where there is potential for unending loops,
which may be the cause of this crash. If it is it's fairly easy fix, but
determining if it's actually going to work is impossible without reliable
reproduction steps.

**Comment by AndrewTheel** [comment]
Apparently a workaround players are using is running windowed mode.

**Comment by cmbasnett** [comment]
I'm not sure how that would affect things, but if it's true and it actually
helps then it potentially points to a lower level problem, or at least a
manifestation of it brought on by the changes to spectator mode.

**Comment by AndrewTheel** [comment]
To get to the bottom of this issue, we should collect multiple user's INIs to determine if there are any common settings among them which create this problem for them and them alone. Also, might be wise to get their GPU, as I have a feeling this is a GPU / graphic issue.

**Comment by AndrewTheel** [comment]
I tried enabling and disabling several graphical options including blur made no difference.

**Comment by Oakbough** [comment]
I just bought the game through Steam. 
I'm enjoying it greatly except the fact that the game and my system freeze upon player death. This makes it completely unplayable. I have tried windowed mode and no motion blur. Please let me know if there is a workaround or how I can help contribute info for a fix
Thanks

**Comment by AndrewTheel** [comment]
@Oakbough Yes, you can be helpful. We need someone with the problem to help us debug it. Could start out with getting a reliable way to replicate the result. We want to know if it can happen in singleplayer with bots. We want a log from a client which experienced the freeze. I'm guessing the log file isn't being created because of the freeze however, so perhaps you might need to play with `showlog` and once the .exe freezes to switch to the log and copy-paste.

Another thing that might be helpful is determining if some types of death don't cause the issue, for instance if you are completely obliterated by tank HE, if it does or does not occur.

We are pretty sure it is something to do with the spectate changes we made a while back, but how it is only affecting a random few players is very strange. Perhaps we should sit down with you a few times on discord and do some testing with you.

**Comment by Oakbough** [comment]
Thank you for the prompt response. It appears as though I did indeed have motion blur activated when experiencing the critical crashes.

I have played a few times since turning motion blue off and it has been more stable. I haven't tried using full screen with motion blur off to see what happens. I was happy to see it at least running and wanted to try the game some more. I am not sure if I can be helpful considering the fix is something you are aware of but if I can be of help I can gladly talk to you about my system and experience with the game.

Thanks
Jean-Paul
________________________________

**Comment by AndrewTheel** [comment]
This seems to be universally a problem with clients using integrated GPU cards. (basically dells, cheap PCs, laptops).

**Comment by AndrewTheel** [comment]
I tested my laptop's integrated GPU and did not have an issue. Still not sure what this problem is, I think we may just have to blind debug it.

**Comment by AndrewTheel** [comment]
Basnett and I decided to have this after v8.2, we will gut it entirely and see if the crash on death issue goes away, if it does then we will consider re-implementing the spectator changes one by one slowly so we can determine where the issue is.

**Comment by AndrewTheel** [comment]
Okay so update here. We instead decided to change 1 thing first before we gut all the spectator changes. I've re-implemented self-spectate mode (aka made it always valid even when the player has no pawn). I have multiple reports that the crash very often happens when the player dies on a gun or when they are gibbed. So by turning on this self-spectate mode we hope the issue goes away. Because otherwise we don't see how any other changes could have changed anything for those players. However, we want DH to be stable for as much hardware as possible and are willing to revert all the spectator changes to see if the crashes go away. If the crashes go away we can pretty much leave it how it currently is, but if they don't we have to go further down the rabbit hole.

**Comment by AndrewTheel** [comment]
Update:
The first hopeful fix did not work. Players still crash constantly when they did. Next up is to revert the HUD changes from the spectator update.

**Comment by AndrewTheel** [comment]
Hmm just thought of something, we should find out if those with issues are running an Anti-Virus and if so, which one.
